library/src/scala/tasty/reflect/CompilerInterface.scala

```diff
@@ -162,6 +162,12 @@ trait CompilerInterface {
   /** Get method symbol if method is either defined in current compilation run or present on classpath. Throws if the method has an overload. */
   def Context_requiredMethod(self: Context)(path: String): Symbol
 
+  /** Get Java class name if we've accidentally tried to reflect on a Java class.  None returned if TASTy class. */
+  def Context_javaCompilationUnitClassname(self: Context): Option[String]
+
+  /** Get Scala class name if attempted reflection is performed on an older Scala file w/o Tasty information present. */
+  def Context_nonTastyScalaCompilationUnitClassname(self: Context): Option[String]
+
```

Comment:
We should have the following API instead. Similar in `Reflection`.
```scala
  def Context_compilationUnitClassname(self: Context): String
  def Context_isJavaCompilationUnit(self: Context): Boolean
  def Context_isScala2CompilationUnit(self: Context): Boolean
```

Reply:
Then your example would be changed to (or you can have extension methods to wrap thing into options on your side):

```scala
  protected def processCompilationUnit(reflect: Reflection)(root: reflect.Tree): Unit = 
    import reflect.{given,_}
    if reflect.rootContext.isJavaCompilationUnit then 
      doSomethingHereWithJava(reflect.rootContext.compilationUnitClassname)
    else 
      doSomethingHereWithTasty(...)
```